sandbox/visualizations_emp.R

```diff
@@ -11,176 +11,368 @@ library(leaflet.providers)
 
-exif_20230528_0008_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230528-0008.csv")
+totalexif4$duration <- as.duration(totalexif4$datetime[1] %--% totalexif4$datetime)
 
-exif_20230528_0008_metadata$X <- 1:nrow(exif_20230528_0008_metadata)
+totalexif5$duration <- as.duration(totalexif5$datetime[1] %--% totalexif5$datetime)
 
-exif_20230528_0009_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230528-0009.csv")
+totalexif6$duration <- as.duration(totalexif6$datetime[1] %--% totalexif6$datetime)
 
-exif_20230528_0009_metadata$X <- 1:nrow(exif_20230528_0009_metadata)
+#### exif1 interactive map ####
 
-exif_20230706_0152_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230706-0152.csv")
+## define functions for continuous coloring
 
-exif_20230706_0152_metadata$X <- 1:nrow(exif_20230706_0152_metadata)
+paletteNumASL1 <- colorNumeric('YlOrRd', domain = totalexif1$altitude_asl)
 
-exif_20230706_0153_metadata <- read.csv ("extracted-metadata\\image-level-metadata\\image-metadata_20230706-0153.csv")
+paletteNumDateTime1 <- colorNumeric('BuPu', domain = totalexif1$duration)
 
-exif_20230706_0153_metadata$X <- 1:nrow(exif_20230706_0153_metadata)
+## the map
 
-## merge original data and image level metadata
+exif1_MAP <-
+  # create leaflet function using data specific to this map
+  leaflet(totalexif1) %>%
+  # use the jawg.terrain basemap
+  addTiles(
+    'https://tile.jawg.io/jawg-terrain/{z}/{x}/{y}{r}.png?access-token=cN5iUzB2bL1QMdeVwMJlflwXQweOoNTDOadGfoCbPf1rRM9FGPb4i1m4f2k5MaJP') %>%
+  # set the center of the map to be the center ish of the flight footprint
+  setView(
+    lng=(totalexif1$lon[888]),
+    lat=(totalexif1$lat[888]),
+    zoom = 16
+    ) %>%
+  # add circles at each image location
+  addCircleMarkers(
+    ~lon,
+    ~lat,
+    popup = paste(
+    "Image filepath:", totalexif1$received_image_path, "<br>",
+    "Collection date and time:", totalexif1$DateTimeOriginal, "<br>",
+    "Drone altitude above sea level:", totalexif1$altitude_asl, " meters",        "<br>"),
+    radius = 2,
+    weight = 1,
+    stroke = TRUE,
+    fillOpacity = 1,
+    color = "#008080"
+    ) %>%
+  # add secret invisible circles at each image location with a bigger radius so they're actually clickable
+  addCircleMarkers(
+    ~lon,
+    ~lat,
+    popup = paste(
+      "Image filepath:", totalexif1$received_image_path, "<br>",
+      "Collection date and time:", totalexif1$DateTimeOriginal, "<br>",
+      "Drone altitude above sea level:", totalexif1$altitude_asl, " meters",        "<br>"),
+    radius = 5,
+    weight = 1,
+    stroke = FALSE,
+    fillOpacity = 0
+  ) %>%
+  # add an interactive layer for ASL
+  addCircleMarkers(
+    ~lon,
+    ~lat,
+    popup = paste(
+      "Image filepath:", totalexif1$received_image_path, "<br>",
+      "Drone altitude above sea level:", totalexif1$altitude_asl, " meters",        "<br>"),
```

Comment:
Can we drop the popup here since it's already included in the larger invisible clickable circles? Same for the collection time circles.

Reply:
done!